src/Wolf.cpp

```diff
@@ -0,0 +1,466 @@
+/*******************************************************************************
+GPU OPTIMIZED MONTE CARLO (GOMC) 2.70
+Copyright (C) 2018  GOMC Group
+A copy of the GNU General Public License can be found in the COPYRIGHT.txt
+along with this program, also can be found at <http://www.gnu.org/licenses/>.
+********************************************************************************/
+#include "Wolf.h"
+#include "Ewald.h"
+#include "CalculateEnergy.h"
+#include "EnergyTypes.h"            //Energy structs
+#include "EnsemblePreprocessor.h"   //Flags
+#include "BasicTypes.h"             //uint
+#include "System.h"                 //For init
+#include "StaticVals.h"             //For init
+#include "Forcefield.h"
+#include "MoleculeKind.h"
+#include "Coordinates.h"
+#include "BoxDimensions.h"
+#include "TrialMol.h"
+#include "GeomLib.h"
+#include "NumLib.h"
+#include <cassert>
+#ifdef GOMC_CUDA
+#include "CalculateEwaldCUDAKernel.cuh"
+#include "CalculateForceCUDAKernel.cuh"
+#include "ConstantDefinitionsCUDAKernel.cuh"
+#endif
+#include "GOMCEventsProfile.h"
+
+
+Wolf::Wolf(StaticVals & stat, System & sys) :
+  Ewald(stat, sys) {
+    for(uint b = 0 ; b < BOX_TOTAL; b++) {
+        wolfAlpha[b] = ff.wolfAlpha[b];
+        wolfFactor1[b] = ff.wolfFactor1[b];
+    }
+}
+
+void Wolf::Init() {
+    for(uint m = 0; m < mols.count; ++m) {
+        const MoleculeKind& molKind = mols.GetKind(m);
+        for(uint a = 0; a < molKind.NumAtoms(); ++a) {
+          particleKind.push_back(molKind.AtomKind(a));
+          particleMol.push_back(m);
+          particleCharge.push_back(molKind.AtomCharge(a));
```

Comment:
Use the variables in MoleculeLookup class. Otherwise will have multiple copy of same data with the size of (N x double) each